### ISSUE TITLE
Table ignores codec configuration upon restart

### ISSUE THREAD

**Issue by seifertm** [robinhood/faust#148]
As far as I can tell, `Table` ignores the codec configuration when re-building the changelog and defaults to `json`. The stack trace shows that Faust is invoking the JSON decoder, even though the app specifies the `raw` codec.

Tested with faust-1.0.30.

## Steps to reproduce
* Run the provided example application
* Restart the example application

```py
import faust


app = faust.App('test', broker='kafka://localhost:9094', store='memory://', key_serializer='raw', value_serializer='raw')


message_topic = app.topic('messages', key_type=bytes, value_type=bytes)
messages = app.Table('messages')


@app.agent(message_topic)
async def build_item_index(stream: faust.Stream):
    async for message in stream:
        messages[b'key'] = message


@app.task
async def on_started():
    await message_topic.send(value=b'Test message')
    for message in messages:
        print(message)
```

## Expected behavior
The application is restarted successfully.

## Actual behavior
The application crashes upon restart.

## Full traceback
```pytb
[2018-08-24 19:30:43,069: ERROR]: [^---ChangelogReader: messages]: Crashed reason=KeyDecodeError('Expecting value: line 1 column 1 (char 0)',)
Traceback (most recent call last):
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/registry.py", line 54, in loads_key
    payload = self._loads(serializer, key)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/registry.py", line 63, in _loads
    return loads(serializer, data)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/codecs.py", line 343, in loads
    return get_codec(codec).loads(s) if codec else s
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/codecs.py", line 223, in loads
    s = cast(Codec, node)._loads(s)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/codecs.py", line 248, in _loads
    return _json.loads(want_str(s))
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/utils/json.py", line 121, in loads
    return json.loads(s, **kwargs)
  File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/mode/services.py", line 688, in _execute_task
    await task
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/mode/services.py", line 417, in _and_transition
    return await fun(self, *args, **kwargs)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/tables/changelogs.py", line 194, in _read
    await self._slurp_stream()
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/tables/changelogs.py", line 215, in _slurp_stream
    async for i, event in aenumerate(self._read_changelog()):
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/mode/utils/aiter.py", line 13, in aenumerate
    async for item in it:
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/tables/changelogs.py", line 242, in _read_changelog
    async for event in self.channel:
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/channels.py", line 301, in __anext__
    return await self.queue.get()
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/mode/utils/queues.py", line 124, in get
    raise self._errors.popleft()
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/topics.py", line 153, in decode
    k = loads_key(key_type, message.key, serializer=key_serializer)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/registry.py", line 60, in loads_key
    sys.exc_info()[2]) from exc
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/registry.py", line 54, in loads_key
    payload = self._loads(serializer, key)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/registry.py", line 63, in _loads
    return loads(serializer, data)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/codecs.py", line 343, in loads
    return get_codec(codec).loads(s) if codec else s
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/codecs.py", line 223, in loads
    s = cast(Codec, node)._loads(s)
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/serializers/codecs.py", line 248, in _loads
    return _json.loads(want_str(s))
  File "/home/michael/arbeit/digitalernachschub/ameto/ameto/api/venv/lib64/python3.6/site-packages/faust/utils/json.py", line 121, in loads
    return json.loads(s, **kwargs)
  File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
faust.exceptions.KeyDecodeError: Expecting value: line 1 column 1 (char 0)
```

# Versions

* Python version: 3.6.5
* Faust version: 1.0.30
* Operating system: Linux
* Kafka version: 1.0.1

**Comment by seifertm** [comment]
Have a look at `faust.tables.base._new_changelog`: https://github.com/robinhood/faust/blob/ee66b1ad428af0876f952b3248900cdc45e4661c/faust/tables/base.py#L321-L322

This always creates a changelog topic with json serialization. The minimum example I provided in this issue runs, if I replace these two lines with the following:

```py
key_serializer=self.app.conf.key_serializer,
value_serializer=self.app.conf.value_serializer,
```

But what happens when the user did not configure serializers in the app?

Hope this helps, but I currently don't know the codebase well enough to compile a proper patch.

**Comment by seifertm** [comment]
I consider this issue fixed as of #208 (tested with faust-1.4.6). Thanks @wuttem and @ask!

For future reference:
Note that the initial example does not work with v1.4.6 without changes. The code configures the key and value types for the topic, but not for the changelog topic. Therefore the changelog topic uses default (JSON) serialization. For the example to work, it is necessary to specify the key and value types for the changelog topic, because we are trying to write bytes to the table in `build_item_index`:
```
messages = app.Table('messages', key_type=bytes, value_type=bytes)
```
